presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -233,26 +332,42 @@ public void finish()
             return;
         }
         finished = true;
+        ImmutableMap.Builder<DynamicFilterId, Domain> domainsBuilder = new ImmutableMap.Builder<>();
         if (valueSets == null) {
-            return; // the predicate became too large.
+            // the predicate became too large
+            if (minValues != null && maxValues != null) {
+                // create TupleDomain from min/max values
+                for (Integer channelIndex : minMaxChannels) {
+                    Type type = channels.get(channelIndex).type;
+                    Object min = readNativeValue(type, minValues[channelIndex], 0);
+                    Object max = readNativeValue(type, maxValues[channelIndex], 0);
+                    Domain domain = Domain.create(ValueSet.ofRanges(
+                            Range.range(type, min, true, max, true)), false);
+                    domainsBuilder.put(channels.get(channelIndex).filterId, domain);
+                }
+                minValues = null;
+                maxValues = null;
+                dynamicPredicateConsumer.accept(TupleDomain.withColumnDomains(domainsBuilder.build()));
+            }
```

Comment:
`return` here, then you can avoid `else` below

Reply:
done